### PR TITLE
[20755] Improve path processing 

### PR DESCRIPTION
**Pull request by elianalf**
This PR introduces improvements regarding path processing:
* Path normalization
* Replace '\' with '/' in dependencies

### CODE REVIEW

File: src/main/java/com/eprosima/idl/context/Context.java [hunk 1]
```diff
@@ -986,6 +982,8 @@ else if (flag == 3)
                                     includeFile.substring(0, includeFile.length() - 4).replace('.', '_');
                                     // Change back to "." for relative paths.
                                     dependency_to_add = dependency_to_add.replaceAll("__/", "../");
+                                    // Substitute "\\" with File separator.
+                                    dependency_to_add = dependency_to_add.replace('\\', '/');
```
Comment (Mario-DL) [inline src/main/java/com/eprosima/idl/context/Context.java:986]: I think that the suggested change on the issue was on a previous line
```suggestion
                                    includeFile.substring(0, includeFile.length() - 4).replace('.', '_').replace('\\', '/'));
                                    // Change back to "." for relative paths.
                                    dependency_to_add = dependency_to_add.replaceAll("__/", "../");
```

**Review by Mario-DL — CHANGES_REQUESTED** [review]

File: src/main/java/com/eprosima/idl/context/Context.java (hunk 1 above)
Comment (Mario-DL) [reply src/main/java/com/eprosima/idl/context/Context.java:986]: Well, I think its the same in this case, as in this recent version, a temporal object is created. Resolving

**Review by Mario-DL — APPROVED** [review]
LGTM, we may need to backport this

**Comment by EduPonz** [comment]
@elianalf can you please study the need of backports

**Comment by elianalf** [comment]
@Mergifyio backport 2.0.x 3.0.x 4.0.x-devel

**Pull request merged**